libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxManeuverView.kt

```diff
@@ -103,12 +106,40 @@ class MapboxManeuverView : ConstraintLayout {
 
     /**
      * Invoke the method to render primary, secondary, sub instructions and lane information.
-     * @param maneuver Expected
+     * @param maneuvers Expected
+     */
+    fun renderManeuver(maneuvers: Expected<ManeuverError, List<Maneuver>>) {
+        maneuvers.fold(
+            { error ->
+                //
+            },
+            { list ->
+                if (list.isNotEmpty()) {
+                    drawManeuver(list[0])
+                }
+            }
+        )
+    }
+
+    /**
+     * Given a list of [Maneuver] and [Map] of id to [RoadShield], the function maintains a
+     * map of id to [RoadShield] that is used to render the shield on the view
+     * @param maneuvers List<Maneuver>
+     * @param shieldMap Map<String, RoadShield?>
      */
-    fun renderManeuver(maneuver: Expected<ManeuverError, Maneuver>) {
-        maneuver.onValue {
-            drawManeuver(it)
+    fun updateRoadShields(maneuvers: List<Maneuver>, shieldMap: Map<String, RoadShield?>) {
```

Comment:
This method cannot be removed. There are 2 use cases:
**Use case 1:**
You want to render the maneuver only after road shield data is readily available. In this case you use this new method
`fun renderManeuver(maneuvers: Expected<ManeuverError, List<Maneuver>>, shieldMap: Map<String, RoadShield?>)`
**Use case 2:**
You want to render text maneuver first and then add road shield to the same maneuver later. In this case you use this method ``fun renderManeuver(maneuvers: Expected<ManeuverError, List<Maneuver>>)` which only renders the text. Now, to render the road shield you would have to maintain the state of road shields as discussed. 

Also, this new method is just for convenience, but mostly won't be used because it depends on the availability of road shields. Using this method would delay the rendering of maneuver by few milliseconds because it will take time to download the shields. It will be more prominent in case of off route and any other changes to route.
cc @LukasPaczos

Reply:
I think I'm still not seeing why do we need
```
fun updateRoadShields(maneuvers: List<Maneuver>, shieldMap: Map<String, RoadShield?>)
```

We could achieve the same result with
```
fun renderManeuvers(maneuvers: Expected<ManeuverError, List<Maneuver>>,shieldMap: Map<String, RoadShield?>)
```

where `updateRoadShields` is even worse because it doesn't actually draw the shields, it only stores the state as far as I can tell. Why would we ever want to use it instead of `renderManeuver(maneuvers, shields)` after the shields are available?

If we don't want to wait for shields, we first use `renderManeuvers(maneuvers)` and then follow up with `renderManeuvers(maneuvers, shields)` when shields are available. What am I missing?

Reply:
If it would help with usability, the `RoadShieldCallback` can always return `Expected<ManeuverError, List<Maneuver>>` instead of `List<Maneuver>` to match other signatures, instead of exposing a new method.